### PR TITLE
nextcloudPackages.whiteboard: init at 1.0.4; nextcloudPackages: update

### PR DESCRIPTION
**Pull request by onny**
## Description of changes

Add whiteboard to package set https://github.com/nextcloud/whiteboard

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/servers/nextcloud/packages/nextcloud-apps.json [hunk 1]
```diff
@@ -37,4 +37,5 @@
 , "unsplash": "agpl3Only"
 , "user_oidc": "agpl3Plus"
 , "user_saml": "agpl3Plus"
+, "whiteboard": "agpl3Only"
```
Comment (dotlambda) [inline pkgs/servers/nextcloud/packages/nextcloud-apps.json:40]:
```suggestion
, "whiteboard": "agpl3Plus"
```
https://github.com/nextcloud/whiteboard/blob/main/package.json#L4

**Comment by dotlambda** [comment]
We should also package the backend and add a service.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by pyrox0** [comment]
I am not a maintainer for this package set, please do not request reviews from me on it.

**Comment by onny** [comment]
@dotlambda Good point, packaged the backend server and added an module for it.

Works for me

``` 
  environment.etc."nextcloud-admin-pass".text = "test123";
  services.nextcloud = {
    enable = true;
    hostName = "localhost";
    extraApps = {
      inherit (config.services.nextcloud.package.packages.apps) whiteboard;
    };
    extraAppsEnable = true;
    config.adminpassFile = "/etc/nextcloud-admin-pass";
  };

  services.nextcloud-whiteboard-server = {
    enable = true;
    settings = {
      NEXTCLOUD_URL = "http://localhost";
      JWT_SECRET_KEY = "test123";
    };
  };
```

```
nextcloud-occ config:app:set whiteboard collabBackendUrl --value="http://localhost:3002"
nextcloud-occ config:app:set whiteboard jwt_secret_key --value="test123"
```

File: nixos/modules/services/web-apps/nextcloud-whiteboard-server.nix [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      '';
+      example = literalExpression ''
+        {
+          NEXTCLOUD_URL = "https://nextcloud.example.org";
+        }
+      '';
+    };
+
+    secrets = lib.mkOption {
+      type = with types; listOf path;
```
Comment (Ma27) [inline nixos/modules/services/web-apps/nextcloud-whiteboard-server.nix:41]: This should really be a `str`.

See https://github.com/NixOS/nixpkgs/pull/326306#discussion_r1675992985 for a little bit of context.

File: nixos/modules/services/web-apps/nextcloud-whiteboard-server.nix [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    systemd.services.nextcloud-whiteboard-server = {
+      description = "Nextcloud backend server for the Whiteboard app";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target" ];
+      wants = [ "network-online.target" ];
+      environment = cfg.settings;
+      serviceConfig = {
+        ExecStart = "${lib.getExe pkgs.nextcloud-whiteboard-server}";
+        WorkingDirectory = "/var/lib/whiteboard";
```
Comment (Ma27) [inline nixos/modules/services/web-apps/nextcloud-whiteboard-server.nix:61]: `%S/whiteboard` may be a little nicer here.

**Review by Ma27 — COMMENTED** [review]
Two notes on the module. Everything else is looking good.

**Comment by onny** [comment]
Thank you @Ma27 fixed it :+1:

**Pull request merged**